pkgs/top-level/haskell-packages.nix

```diff
@@ -82,6 +83,8 @@ in
 
       ghc963Binary = callPackage ../development/compilers/ghc/9.6.3-binary.nix { };
 
+      ghc966Binary = callPackage ../development/compilers/ghc/9.6.6-binary.nix { };
```

Comment:
I think we should clarify that this is not a normal bindist and call this e.g. `ghc966DebianBinary` for clarity. This also frees us from packaging the `.deb` and bindists in the same derivation if we want to package official 9.6.6 bindists later.

Reply:
ACK, will do the rename once I've checked off everything else.